sql-data-pipeline/src/main/java/io/cdap/pipeline/sql/app/core/AbstractSQLExecutor.java

```diff
@@ -0,0 +1,402 @@
+
+    // Flag to enable or disable creation of tables when they don't exist
+    createTableFeatureFlag = false;
+    // Set whether we want to create tables if they do not exist
+    if (getContext().getRuntimeArguments().get(CREATE_TABLE_FEATURE_FLAG) == "true") {
+      createTableFeatureFlag = true;
+    }
+
+    // Load the dag and other properties previously initialized at configure time
+    loadExecutorState(gson, getContext().getSpecification().getProperties());
+
+    // Load the Calcite JDBC driver
+    loadCalciteJdbcDriver();
+    // Traversal Logic to prepare the query ordering
+    traverseTopologicalOrder();
+
+    // Serialize all sink and splitter RelNodes
+    // Converter to convert RelNodes to SqlNodes
+    RelToSqlConverter converter = new RelToSqlConverter(getDialect());
+    // Creates a temporary table for each splitter
+    createTemporaryTables(converter);
+    // Creates a query for each sink
+    createSinkQueries(converter);
+    // Make the query list unmodifiable
+    queries = Collections.unmodifiableList(queries);
+  }
+
+  /**
+   * Initializes the state of the executor and stores the GSON of the state in the properties map. This should
+   * only happen at configure time.
+   */
+  private void initExecutorState(Gson gson, Map<String, String> properties) {
+    // Store the config
+    properties.put(CONFIG_NAME, gson.toJson(config));
+
+    // Initialize the dag
+    if (!config.getConnections().isEmpty()) {
+      dag = new Dag(config.getConnections());
+    } else {
+      dag = null;
+    }
+
+    // Set the connections maps
+    for (Connection connection: config.getConnections()) {
+      if (!fromNodeMap.containsKey(connection.getTo())) {
+        fromNodeMap.put(connection.getTo(), new HashSet<>());
+      }
+      if (!toNodeMap.containsKey(connection.getFrom())) {
+        toNodeMap.put(connection.getFrom(), new HashSet<>());
+      }
+      fromNodeMap.get(connection.getTo()).add(connection.getFrom());
+      toNodeMap.get(connection.getFrom()).add(connection.getTo());
+    }
+    // Save the connections maps
+    properties.put(FROM_CONNECTIONS_NAME, gson.toJson(fromNodeMap));
+    properties.put(TO_CONNECTIONS_NAME, gson.toJson(toNodeMap));
+
+    // Set sources and sinks
+    for (ETLStage stage: config.getStages()) {
+      if (!fromNodeMap.containsKey(stage.getName())) {
+        sourceStages.add(stage.getName());
+      }
+      if (!toNodeMap.containsKey(stage.getName())) {
+        sinkStages.add(stage.getName());
+      }
+    }
+    // Save the sources and sinks
+    properties.put(SOURCES_NAME, gson.toJson(sourceStages));
+    properties.put(SINKS_NAME, gson.toJson(sinkStages));
+
+    // Store the config JSON for access during runtime
+    setProperties(properties);
+  }
+
+  /**
+   * Loads the executor state property GSONs from the property map. This should only happen at runtime.
+   */
+  private void loadExecutorState(Gson gson, Map<String, String> properties) {
```

Comment:
Just call this `loadExecutor()` or `loadExecutorConfig()`, I don't see there is any state involved

Reply:
Fixed